nav2_controller/include/nav2_controller/nav2_controller.hpp

```diff
@@ -34,39 +34,126 @@ namespace nav2_controller
+  /**
+   * @brief Activates member variables
+   *
+   * Activates controller, costmap, velocity publisher and follow path action
+   * server
+   * @param state LifeCycle Node's state
+   * @return Success or Failure
+   */
   nav2_util::CallbackReturn on_activate(const rclcpp_lifecycle::State & state) override;
+  /**
+   * @brief Deactivates member variables
+   *
+   * Deactivates follow path action server, controller, costmap and velocity
+   * publisher. Before calling deactivate state, velocity is being set to zero.
+   * @param state LifeCycle Node's state
+   * @return Success or Failure
+   */
   nav2_util::CallbackReturn on_deactivate(const rclcpp_lifecycle::State & state) override;
+  /**
+   * @brief Calls clean up states and resets member variables.
+   *
+   * Controller and costmap clean up state is called, and resets rest of the
+   * variables
+   * @param state LifeCycle Node's state
+   * @return Success or Failure
+   */
   nav2_util::CallbackReturn on_cleanup(const rclcpp_lifecycle::State & state) override;
+  /**
+   * @brief Called when in Shutdown state
+   * @param state LifeCycle Node's state
+   * @return Success or Failure
+   */
   nav2_util::CallbackReturn on_shutdown(const rclcpp_lifecycle::State & state) override;
+  /**
+   * @brief Called when in Error state
+   * @param state LifeCycle Node's state
+   * @return Success or Failure
+   */
   nav2_util::CallbackReturn on_error(const rclcpp_lifecycle::State & state) override;
 
   using ActionServer = nav2_util::SimpleActionServer<nav2_msgs::action::FollowPath>;
 
   // Our action server implements the FollowPath action
   std::unique_ptr<ActionServer> action_server_;
-
-  // The action server callback
+  /**
+   * @brief FollowPath action server callback. Handles action server updates and
+   * spins server until goal is reached
+   *
+   * Provides global path to controller received from action client. Twist
+   * velocities for the robot are calculated and published using controller at
+   * the specified rate till the goal is reached.
+   * @throw nav2_core::PlannerException
+   */
   void followPath();
-
+  /**
+   * @brief Assigns path to controller
+   * @param path Path received from action server
+   */
   void setPlannerPath(const nav_msgs::msg::Path & path);
+  /**
+   * @brief Calculates velocity and publishes to "cmd_vel" topic
+   */
   void computeAndPublishVelocity();
+  /**
+   * @brief Calls setPlannerPath method with an updated path received from
+   * action server
+   */
   void updateGlobalPath();
+  /**
+   * @brief Calls velocity publisher to publish the velocity on "cmd_vel" topic
+   * @param velocity Twist velocity to be published
+   */
   void publishVelocity(const geometry_msgs::msg::TwistStamped & velocity);
+  /**
+   * @brief Calls velocity publisher to publish zero velocity
+   */
   void publishZeroVelocity();
+  /**
+   * @brief Checks if robot reached goal position by calling getRobotPose method
```

Comment:
I think you're being too literal here -- what's the _purpose_ behind this method and what's the goal it accomplishes? It doesn't check the position, its checking to see if the goal is reached.

Reply:
I totally agree, but I was addressing your "how does it checks that?" comment.

Reply:
Will this be fine?
```
/**
   * @brief Checks if goal is reached
   * @return true or false
   */
```

Reply:
Works